docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,421 @@
+---
+id: higher-order-components
+title: Higher-Order Components
+permalink: docs/higher-order-components.html
+---
+
+A higher-order component (HOC) is an advanced technique in React for reusing component logic. HOCs are not part of the React API, per se. They are a pattern that emerges from React's compositional nature.
+
+Concretely, **a higher-order component is a function that takes a component and returns a new component.**
+
+```js
+const EnhancedComponent = higherOrderComponent(WrappedComponent);
+```
+
+Whereas a component transforms props into UI, a higher-order component transforms themselves.
```

Comment:
Lol that's what I had before the last commit and @gaearon asked me to change it to this :D https://github.com/facebook/react/pull/7869#discussion_r82143805